apps/obsidian/src/components/SearchBar.tsx

```diff
@@ -0,0 +1,224 @@
+import { App, debounce } from "obsidian";
+import React, { useEffect, useState, useMemo, ReactNode } from "react";
+import { QueryEngine } from "~/services/QueryEngine";
+
+type SearchBarBaseProps<T> = {
+  app: App;
+  placeholder?: string;
+  onNodeSelect: (item: T | null) => void;
+  debounceMs?: number;
+  minQueryLength?: number;
```

Comment:
These aren't being customized by any implementations so let's just set them in the function and remove them from props.

Let's add additional props only when required.